lib/Dialect/FIRRTL/InstanceGraph.cpp

```diff
@@ -12,14 +12,42 @@
 using namespace circt;
 using namespace firrtl;
 
-InstanceRecord *InstanceGraphNode::recordInstance(InstanceOp instance,
-                                                  InstanceGraphNode *target) {
-  moduleInstances.emplace_back(instance, this, target);
-  return &moduleInstances.back();
+void InstanceRecord::erase() {
```

Comment:
I'm surprised llvm doesn't have a helper class you can inherit from that carries with it all the intrusive pointers and api to simplify simple use cases.